src/lib.rs

```diff
@@ -406,6 +414,42 @@ impl PurpleICQ {
         }
     }
 
+    pub fn serv_got_chat_in(&mut self, connection: &mut Connection, chat_info: MsgInfo) {
+        match purple::Chat::find(&mut connection.get_account(), &chat_info.chat_sn) {
+            Some(mut chat) => {
+                let mut node = chat.as_blist_node();
+                let last_timestamp: i64 = {
+                    match node.get_string(&blist_node::LAST_SEEN_TIMESTAMP) {
+                        Some(timestamp) => timestamp.parse::<i64>().unwrap_or(0),
+                        None => 0,
+                    }
+                };
```

Comment:
I feel like there is a one-liner for this so that we don't have to set the default twice. I just can't find it.

Reply:
`node.get_string(&blist_node::LAST_SEEN_TIMESTAMP).and_then(|t| t.parse::<i64>()).unwrap_or(0)`

Reply:
Ahh nice. Thanks!